packages/client/src/runtime/core/engines/library/DefaultLibraryLoader.ts

```diff
@@ -187,21 +187,10 @@ Read more about deploying Prisma Client: https://pris.ly/d/client-generator`
       return { enginePath, searchedLocations }
     }
 
-    const dirname = eval('__dirname') as string
     const searchLocations: string[] = [
-      // TODO: why hardcoded path? why not look for .prisma/client upwards?
-      path.resolve(dirname, '../../../.prisma/client'), // Dot Prisma Path
-      this.config.generator?.output?.value ?? dirname, // Custom Generator Path
-      path.resolve(dirname, '..'), // parentDirName
-      path.dirname(this.config.datamodelPath), // Datamodel Dir
-      this.config.cwd, //cwdPath
-      '/tmp/prisma-engines',
+      path.dirname(this.config.datamodelPath),
```

Comment:
```suggestion
      this.config.dirname,
```
Code says it can be undefined, but is always present.

Reply:
This with what we got the data model path in the first place

Reply:
That would be the (removed) lines 201-203...
Why could we currently be using both?

Reply:
Maybe the logic was different a long time ago, I cannot tell for sure.

Reply:
Comes from here https://github.com/prisma/prisma/blob/ea22c620c075a244734ff38c8ad66cbbe8538800/packages/client/src/runtime/getPrismaClient.ts/#L439